File: ABB-setup/dv-processing-rel_1.7/docs/source/assets/code_samples/python/reading_data/sample5.py
import time
import dv_processing as dv

# Open any camera
capture = dv.io.CameraCapture()

# Run the loop while camera is still connected
while capture.isRunning():
    # Read a batch of IMU data from the camera
    imu_batch = capture.getNextImuBatch()

    # The method does not wait for data to arrive, it returns immediately with
    # latest available data or if no data is available, returns a `None`
    if imu_batch is not None and len(imu_batch) > 0:
        # Print the time range of imu data
        print(f"Received imu data within time range [{imu_batch[0].timestamp}; {imu_batch[-1].timestamp}]")
    else:
        time.sleep(0.001)

File: ABB-setup/dv-processing-rel_1.7/docs/source/assets/code_samples/c++/reading_data/sample5.cpp
#include <dv-processing/io/camera_capture.hpp>

#include <chrono>

int main() {
	using namespace std::chrono_literals;

	// Open any camera
	dv::io::CameraCapture capture;

	// Run the loop while camera is still connected
	while (capture.isRunning()) {
		// Read IMU measurement batch, check whether it is correct.
		// The method does not wait for data to arrive, it returns immediately with
		// the latest available imu data or if no data is available, returns a `std::nullopt`.
		if (const auto imuBatch = capture.getNextImuBatch(); imuBatch.has_value() && !imuBatch->empty()) {
			std::cout << "Received " << imuBatch->size() << " IMU measurements" << std::endl;
		}
		else {
			// No data has arrived yet, short sleep to reduce CPU load.
			std::this_thread::sleep_for(1ms);
		}
	}

	return 0;
}
